### ISSUE TITLE
Wrong check for pdftex inside sphinx.sty (for graphicx package option)

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#2164]
When using `make latex`,  one finds in `build/latex` a file `sphinx.sty`. This LaTeX style file is loaded by the produced document. It executes

``` latex
% For graphicx, check if we are compiling under latex or pdflatex.
\ifx\pdftexversion\undefined
  \usepackage{graphicx}
\else
  \usepackage[pdftex]{graphicx}
\fi
```

However, on current main `TeX` installations, `\pdftexversion` is defined also when running `latex` to produce `dvi` files. Thus always the "`pdftex`" branch is chosen, which makes it a priori impossible to compile the document via `latex` (for later producting the `pdf` via `dvipdfmx` for example).

Furthermore,  `graphicx` is smart enough to detect `pdftex` (correctly ... `;-)`), thus the test is not needed.

Hence, I think the `sphinx.sty` should only do `\usepackage{graphicx}`.

**Comment by xuhdev** [comment]
I agree with this change. Actually this detection makes me force to `\usepackage{svg}` after sphinx package in #2166

**Comment by jfbu** [comment]
@xuhdev there is another reason why `\usepackage[pdftex]{graphicx}` is bad, even when compiling with `pdflatex`. Another package added to the preamble _before_  `\usepackage{sphinx}` (via the `'fontenc'` key for example) could very well do `\usepackage{graphicx}`. One then get an error:

``` latex
./testgraphicx.tex:6: LaTeX Error: Option clash for package graphicx.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...                                              

l.6 

The package graphicx has already been loaded with options:
  []
There has now been an attempt to load it with options
  [pdftex]
Adding the global options:
  ,pdftex
to your \documentclass declaration may fix this.
```

This happened to me because I wanted to compile the document to `pdf` via `lualatex`, using system fonts. I thus used the `'fontenc'` key of `latex_elements` in `conf.py` to do a `\usepackage{fontspec}`. But this package at some point (or rather `xunicode.sty` which it loads) does `\usepackage{graphicx}`. The fix for this was to insert also `\PassOptionsToPackage{pdftex}{graphicx}` with the `'fontenc'` key.

Another minor issue is that `sphinx.sty` as a package should not do `\usepackage` but `\RequirePackage`, although the difference is quasi-cosmetic. See

http://tex.stackexchange.com/questions/19919/whats-the-difference-between-requirepackage-and-usepackage

particularly http://tex.stackexchange.com/a/19933/4686

**Comment by xuhdev** [comment]
This problem also bugs me a lot (can't use `latex` to compile). @jfbu Would you mind making a pull request to fix this issue (PR usually gets more attention than an issue :) )? The driver detection configuration in TeXLive is [this](http://texdoc.net/texmf-dist/tex/latex/latexconfig/graphics.cfg), and MiKTeX also includes another version of this auto detection file, thus I don't think sphinx should try to make the guess while the TeX distributions have already done a pretty good work.

**Comment by jfbu** [comment]
@xuhdev I have done the pull request. Not being used to this, I wasn't sure I had to fork the whole [sphinx-doc/sphinx](https://github.com/sphinx-doc/sphinx) to my account, but this is how I proceeded in the end.

**Comment by tk0miya** [comment]
Merged into master branch (at 297c1647110b766c6d0aeadcbaa74a365d75d2a1).
Thanks.
